diff --git a/apps/openmw/engine.cpp b/apps/openmw/engine.cpp
index 0de3f5e29..0d766d14d 100644
--- a/apps/openmw/engine.cpp
+++ b/apps/openmw/engine.cpp
@@ -545,6 +545,9 @@ void OMW::Engine::setSkipMenu (bool skipMenu, bool newGame)
     mNewGame = newGame;
 }
 
+// To share the viewer with Android interfaces
+osg::ref_ptr<osgViewer::Viewer> g_viewer;
+
 void OMW::Engine::createWindow()
 {
     int screen = Settings::Manager::getInt("screen", "Video");
@@ -681,6 +684,9 @@ void OMW::Engine::createWindow()
     mViewer->realize();
 
     mViewer->getEventQueue()->getCurrentEventState()->setWindowRectangle(0, 0, graphicsWindow->getTraits()->width, graphicsWindow->getTraits()->height);
+
+    // To share the viewer with Android interfaces
+    g_viewer = mViewer;
 }
 
 void OMW::Engine::setWindowIcon()
@@ -1109,6 +1115,8 @@ void OMW::Engine::go()
 
     luaWorker.join();
 
+    g_viewer.release();
+
     // Save user settings
     Settings::Manager::saveUser((mCfgMgr.getUserConfigPath() / "settings.cfg").string());
     Settings::ShaderManager::get().save();
diff --git a/components/debug/debugging.cpp b/components/debug/debugging.cpp
index c46971d65..b34bc7a3c 100644
--- a/components/debug/debugging.cpp
+++ b/components/debug/debugging.cpp
@@ -13,6 +13,8 @@
 #include <components/windows.hpp>
 #endif
 
+#include <osgViewer/Viewer>
+
 namespace Debug
 {
 #ifdef _WIN32
@@ -188,6 +190,8 @@ void setupLogging(const std::string& logDir, const std::string& appName, std::io
 #endif
 }
 
+extern osg::ref_ptr<osgViewer::Viewer> g_viewer;
+
 int wrapApplication(int (*innerApplication)(int argc, char *argv[]), int argc, char *argv[],
                     const std::string& appName, bool autoSetupLogging)
 {
@@ -227,6 +231,7 @@ int wrapApplication(int (*innerApplication)(int argc, char *argv[]), int argc, c
     }
     catch (const std::exception& e)
     {
+        g_viewer.release();
 #if (defined(__APPLE__) || defined(__linux) || defined(__unix) || defined(__posix))
         if (!isatty(fileno(stdin)))
 #endif
